dialer.go

```diff
@@ -538,11 +564,11 @@ func (d *Dialer) Close() error {
 	default:
 	}
 	close(d.closed)
-	d.lock.Lock()
-	defer d.lock.Unlock()
-	for _, i := range d.cache {
-		i.Close()
-	}
+
+	d.cache.replaceAll(func(_ instance.ConnName, c *monitoredCache) (instance.ConnName, *monitoredCache) {
```

Comment:
What will this replaceAll function do? Right now I see it's not using any closure state.